src/runtime/opencl.cpp

```diff
@@ -11,9 +11,8 @@ namespace Runtime {
 namespace Internal {
 namespace OpenCL {
 
-// Define the function pointers for the OpenCL API. OpenCL 1.2
-// currently disabled so we can work on build bots without it.
-//#define HAVE_OPENCL_12
+// Define the function pointers for the OpenCL API.
+#define HAVE_OPENCL_12
```

Comment:
Is it safe for us to require opencl 1.2 now?

Reply:
I'm not sure what was intended before, but there were 1.2-introduced functions (clEnqueueWriteImage and company) defined in mini_cl.h and not guarded by this define.  So I think it de facto already required 1.2 (unless you bypassed halide_default_error)

Reply:
And thinking about it, if those API functions are not available I think you won't have an issue if you aren't using the GPUTexture code paths